packages/tradershub/src/features/cfd/screens/EnterPassword/EnterPassword.tsx

```diff
@@ -1,81 +1,61 @@
 import React, { ChangeEvent } from 'react';
-import { useRegulationFlags } from '@/hooks';
-import { TMarketTypes, TPlatforms } from '@/types';
-import { validPassword } from '@/utils';
-import { MarketTypeDetails, PlatformDetails } from '@cfd/constants';
+import { useQueryParams, useRegulationFlags } from '@/hooks';
+import { useCFDContext } from '@/providers';
+import { Category, CFDPlatforms, MarketType, MarketTypeDetails, PlatformDetails } from '@cfd/constants';
 import { useActiveTradingAccount } from '@deriv/api-v2';
-import { Button, PasswordInput, Text, useDevice } from '@deriv-com/ui';
+import { Modal, PasswordInput, Text } from '@deriv-com/ui';
+import MT5PasswordFooter from '../../modals/MT5PasswordModal/MT5PasswordFooter';
 
 type TEnterPasswordProps = {
     isLoading?: boolean;
-    marketType: TMarketTypes.CreateOtherCFDAccount;
     onPasswordChange?: (e: ChangeEvent<HTMLInputElement>) => void;
     onPrimaryClick?: () => void;
     onSecondaryClick?: () => void;
     password: string;
     passwordError?: boolean;
-    platform: TPlatforms.All;
 };
 
 /**
  * Component to display the enter password screen
- * @param {boolean} isLoading - loading state
- * @param {TMarketTypes.CreateOtherCFDAccount} marketType - market type all or synthetic or financial
  * @param {Function} onPasswordChange - callback to handle password change
- * @param {Function} onPrimaryClick - callback to handle primary button click
- * @param {Function} onSecondaryClick - callback to handle secondary button click
  * @param {string} password - password value
- * @param {boolean} passwordError - password error state
- * @param {TPlatforms.All} platform - platform Mt5 or Dxtrade
- * @returns {ReactNode} - returns the enter password screen component
  */
 
-const EnterPassword = ({
-    isLoading,
-    marketType,
-    onPasswordChange,
-    onPrimaryClick,
-    onSecondaryClick,
-    password,
-    passwordError,
-    platform,
-}: TEnterPasswordProps) => {
-    const { isDesktop } = useDevice();
-    const title = PlatformDetails[platform].title;
+const EnterPassword = ({ onPasswordChange, password }: TEnterPasswordProps) => {
     const { isEU } = useRegulationFlags();
-
+    const { cfdState } = useCFDContext();
     const { data } = useActiveTradingAccount();
-    const accountType = data?.is_virtual ? 'Demo' : 'Real';
-    const marketTypeDetails = MarketTypeDetails(isEU);
+    const { closeModal } = useQueryParams();
 
+    const marketTypeDetails = MarketTypeDetails(isEU);
+    const { marketType = MarketType.ALL, platform = CFDPlatforms.MT5 } = cfdState;
+    const accountType = data?.is_virtual ? Category.DEMO : Category.REAL;
+    const title = PlatformDetails[platform].title;
```

Comment:
if you were just renaming the whole thing as title I suggest to destructure it instead.

Reply:
xund